scripts/cf_subshell_scoped_login.sh

```diff
@@ -1,7 +1,9 @@
-#!/bin/sh
+#!/bin/bash
 
 set -eu
 
+[[ "$0" = /* ]] && echo "${0#$PWD}" >> ~/.paas-script-usage || echo "$0" >> ~/.paas-script-usage
```

Comment:
This is pretty cryptic - I'm not even sure what the `[[ "$0" = /* ]]` condition does.

Would `"${0#$PWD}"` not be better as `"$(basename "$0")"`?

Reply:
Ah, I see - you specifically want the path from the current directory to the executable, not just the executable's name.

Reply:
Yeah, I wasn't super happy with this either. The aim was to print out the path of the file relative to the paas-cf directory, when $0 could be anything from just the name of the file (excluding path), to the absolute path to the file.

It reads as
* `[[ "$0" = /* ]] `: does `$0` begin with `/`
* `${0#$PWD}`: strip the value of `$PWD` from the front of `$0`
* `echo $0`: otherwise just `$0`

Is there better way to do this in Bash?

Reply:
I think this could just be:

```bash
echo "${0#$PWD}" >> ~/.paas-script-usage
```

I think there's no need for the conditionals because it will do the right thing anyway:

```bash
BANANA=BANANA
echo "${BANANA#BA}"
# => NANA
echo "${BANANA#BADGER}"
# => BANANA
```

No?